src/views/v2/components/icons/list.php

```diff
@@ -15,9 +15,11 @@
  *
  */
 $svg_classes = [ 'tribe-common-c-svgicon', 'tribe-common-c-svgicon--list' ];
+$clip0 = rand();
+$clip1 = rand();
```

Comment:
We do not want to randomly set an id. We want to set an id but make sure it's not going to clash with other icons. Let's add a sensible id, maybe something related to the class (`tribe-common-c-svgicon__list-clip-path-0`) or something similar.

Reply:
What's happening here is that the icon appears twice in the list view hence why we're getting that replication. We need the IDs for both instances to be unique in this scenario.

Reply:
Is there a better way to ensure the IDs are unique without generating them randomly?

Reply:
So...
Part of the issue is that this exact svg get duplicated on the page - once for mobile and once for desktop.
Any hard-coded string is going to get duplicated. So that won't work.

...however...

We do pass a unique class to this icon here: https://github.com/the-events-calendar/the-events-calendar/blob/master/src/views/v2/components/events-bar/views.php#L37
and here: https://github.com/the-events-calendar/the-events-calendar/blob/master/src/views/v2/components/events-bar/views/list/item.php#L31

Perhaps we can "get sneaky" and use that class - and if we need to we can use a hardcoded pair for a fallback?

Something like 
```php
if ( ! empty( $classes ) ) {
	$svg_classes = array_merge( $svg_classes, $classes ); // this is already in place
	// add more here - grab the first passed class and use that as our ID prefix.
	$id_prefix = array_shift( $classes );
} else {
	$id_prefix = 'tribe-common-c-svgicon__list-clip-path-';
}
```

Then use that prefix to generate two IDs

Reply:
@Camwyn  Great! this approach works well. Here's an updated screenshot of the section 
![image](https://user-images.githubusercontent.com/22029087/135913480-d67fb0d0-c598-4cac-bcd0-5dc38acf127d.png)

Reply:
Also, I've checked and I don't see anywhere we're currently targeting the IDs in CSS and JS.